# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1251402591 14400
#      Thu Aug 27 15:49:51 2009 -0400
# Node ID 8dc1531e214922d9d49b63b1d1dd8e62c549a253
# Parent  bdcfb756d721d173e387cebc1724237f656e98da
correctly save and restore LIBS and CPPFLAGS when checking for GraphicsMagick++

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2009-08-27  John W. Eaton  <jwe@octave.org>
+
+	* configure.in: Properly save and restore CPPFLAGS and LIBS when
+	checking for GraphicsMagick++.  Don't temporarily change LDFLAGS.
+
 2009-08-27  Jaroslav Hajek  <highegg@gmail.com>
 
 	* aclocal.m4 (m4_ifblank, m4_ifnblank): State FSF copyright and markup
 	as FIXME.
 
 2009-08-26  John W. Eaton  <jwe@octave.org>
 
 	* configure.in: Restore LIBS after checking for ARPACK.
diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -580,34 +580,31 @@ PKG_CHECK_EXISTS([GraphicsMagick++], [
   ## GraphicsMagick++ extentions. These extra flags break the Octave build.
   MAGICK_LDFLAGS=`$PKG_CONFIG --libs-only-L GraphicsMagick++`
   MAGICK_LIBS=`$PKG_CONFIG --libs-only-l GraphicsMagick++`
   MAGICK_CPPFLAGS=`$PKG_CONFIG --cflags-only-I GraphicsMagick++`
 
   warn_magick="GraphicsMagick++ library fails tests.  The imread function for reading image files will not be fully functional."
 
   save_CPPFLAGS="$CPPFLAGS"
-  save_LDFLAGS="$LDFLAGS"
   save_LIBS="$LIBS"
-  CPPFLAGS="$CPPFLAGS $MAGICK_CPPFLAGS"
-  LDFLAGS="$LIBS $MAGICK_LDFLAGS"
-  LIBS="$LIBS $MAGICK_LIBS"
+  CPPFLAGS="$MAGICK_CPPFLAGS $CPPFLAGS"
+  LIBS="$MAGICK_LDFLAGS $MAGICK_LIBS $LIBS"
   AC_LANG_PUSH(C++)
     AC_CHECK_HEADER([Magick++.h], [
       AC_MSG_CHECKING([for Magick::ColorRGB in Magick++.h])
       AC_TRY_LINK([#include <Magick++.h>], [Magick::ColorRGB c;], [
         AC_MSG_RESULT(yes)
         warn_magick=
       ], [
         AC_MSG_RESULT(no)
       ])
     ])
   AC_LANG_POP(C++)
   CPPFLAGS="$save_CPPFLAGS"
-  LDFLAGS="$save_LDFLAGS"
   LIBS="$save_LIBS"
 ])
 
 if test -z "$warn_magick"; then
   AC_DEFINE(HAVE_MAGICK, 1, [Define if GraphicsMagick++ is available.])
 else
   MAGICK_CPPFLAGS=
   MAGICK_LDFLAGS=
